### DISCUSSION TITLE
feature: expand the configuration of telescope.

### DISCUSSION THREAD

**Discussion by xieyanknockout** [LazyVim/LazyVim#1627] [Q&A] [answered]
### Did you check the docs?

- [X] I have read all the LazyVim docs

### Is your feature request related to a problem? Please describe.

When using Telescope's livegrep function, I want to filter out some folders. The configuration method I found online is to configure vimgrep_arguments in Telescope's defaults, but I seem unable to add this configuration through opts.

```
defaults = {
    vimgrep_arguments = { -- :h telescope.defaults.vimgrep_arguments for more info
            'rg',

            '--your flags go here',
            '--as comma separated strings',
            '--think about how you want find files to behave, look at ripgrep documentation, add those flags here,'
                      
            -- some examples
            '--hidden',
            '--glob',  -- this flag allows you to hide exclude these files and folders from your search 👇
            '!{**/.git/*,**/node_modules/*,**/package-lock.json,**/pkg/mod/*}', 
                }
        },
```

### Describe the solution you'd like

wish to configure live_grep arguments through opts.

### Describe alternatives you've considered

replace live_grep keys setting?

**Comment by dpetka2001** [comment]
I believe there's a plugin for that, [telescope-live-grep-args.nvim](https://github.com/nvim-telescope/telescope-live-grep-args.nvim). I haven't tried it, but you might want to check that out. Also this is not a Lazyvim issue and is more appropriate for the `Discussions` in my opinion.

**Reply by xieyanknockout** [reply, marked as answer]
thanks for your suggestion, i solved this problem by this configuration
```
-- {$HOME}/.config/nvim/lua/plugins/telescope.lua
return {
	"nvim-telescope/telescope.nvim",
	opts = {
		defaults = {
			vimgrep_arguments = {
				"rg",
				"--color=never",
				"--no-heading",
				"--line-number",
				"--column",
				"--smart-case",
				"--hidden",
				"--glob",
				"!{**/.git/*,**/node_modules/*,**/pkg/mod/*,**/yarn.lock,**/.local/*}",
			},
		},
	},
}

```
